### PR TITLE
Added images on Feed and Saved Fragment

### LINKED ISSUE #15: Add images to FeedFragment [ 1-1.5 week(s) ]
- See [this link](https://fb.quip.com/qM5hARDYvQdn) to view the scope
- Get images for detail page -> save image as Parse file -> load saved Parse image along with other saved show data -> determine API call chain -> load to main Feed RV
- Technically challenging portion!

### PR DESCRIPTION
**Pull request by alexisechano**
- "Brute force" approach to adding images by updating the adapter with new images by locating the show with the show ID
- Changed CardView layout to properly display image
- Updated Parse to save the correct image URL and to load it properly when called in SavedFragment
- Removed image API call and callback from DetailFragment

Screen Shot 2021-07-23 at 2 48 37 PM

https://user-images.githubusercontent.com/16446126/126846466-a93303b7-ffa0-4a9e-bff9-5df4c8ac09ce.mov

### CODE REVIEW

File: app/src/main/java/com/example/next_show/adapters/ShowAdapter.java [hunk 1]
```diff
@@ -55,6 +56,23 @@ public void clear() {
...
 
+    // uses basic search algorithm
+    public int search(String id){
+        for (int index = 0; index < getItemCount(); index++){
+            Show s = shows.get(index);
+            if(s.getId().equals(id)){
+                return index;
+            }
+        }
+        return -1;
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/adapters/ShowAdapter.java:67]: nit: we could save this into a constant like NOT_FOUND

File: app/src/main/java/com/example/next_show/fragments/FeedFragment.java [hunk 2]
```diff
@@ -122,6 +144,24 @@ public boolean onNavigationItemSelected(@NonNull MenuItem item) {
...
+                public void onSuccess(int i, Headers headers, JSON json) {
+                    callback.onSuccess(json);
+                }
+
+                @Override
+                public void onFailure(int i, Headers headers, String s, Throwable throwable) {
+                    callback.onFailure(i);
+                }
+            });
+    }
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/fragments/FeedFragment.java:163]: Do you think we could pull this out into a different class?  And then, we can just reuse this same call on the movies detail screen right?

File: app/src/main/java/com/example/next_show/fragments/FeedFragment.java [hunk 3]
```diff
@@ -204,4 +255,33 @@ public void onFailure(int code) {
...
+
+            // get results and save to show object
+            try {
+                String path = jsonObj.getString("backdrop_path");
+
+                // finds show in adapter
+                int index = adapter.search(jsonObj.getString("id"));
+
+                // updates the show images
+                adapter.updateShowImage(index, path);
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/fragments/FeedFragment.java:275]: Can you check something -- is the list of shows you have line 59/61 up to date with the new image information for the show or do you just update the adapter?

File: app/src/main/java/com/example/next_show/adapters/ShowAdapter.java (hunk 1 above)
Comment (nicholeic) [reply app/src/main/java/com/example/next_show/adapters/ShowAdapter.java:67]: There also might be some api on the adapter itself that you could utilize -- can be a follow up diff, but can you take a look?

File: app/src/main/java/com/example/next_show/adapters/ShowAdapter.java [hunk 4]
```diff
@@ -55,6 +56,23 @@ public void clear() {
...
+    }
+
+    public void updateShowImage(int i, String imgPath) {
+        Show show = shows.get(i);
+        show.setImageUrl(imgPath);
+        notifyItemChanged(i);
+    }
```
Comment (nicholeic) [inline app/src/main/java/com/example/next_show/adapters/ShowAdapter.java:74]: Hm, not sure f this makes sense here -- this is only updating the list here, right?  But you also have this list in the fragment -- I think you might want to do this in the fragment to ensure that list is up to date, and then just tell the adapter that the item changed at the specific index

**Review by nicholeic — COMMENTED** [review]
Some comments -- you can address these in smaller stacked PRs, since this gets the basic code working.

One question, what do you do if an image is already fetched when you get to the details page?  What happens if you go from the feed page to details before the image has been fetched?

File: app/src/main/java/com/example/next_show/fragments/FeedFragment.java (hunk 2 above)
Comment (alexisechano) [reply app/src/main/java/com/example/next_show/fragments/FeedFragment.java:163]: I have no more API call in the details page actually! Although, if the image on the FeedFragment does not load by the time I navigate to the page then there is no image loaded

**Pull request merged**